File: mock-server/src/prescriptions/dto/prescription.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class CreatePrescriptionsDto {

  @ApiProperty({required:false})
  recordNumber: string;

  @ApiProperty({required:false})
  visitDate: string;

  @ApiProperty({required:false})
  diagnosis: string;

  @ApiProperty({required:false})
  pin: string;
  
  @ApiProperty({required:false})
  patient:object;

  @ApiProperty({required:false})
  doctor:object;

  @ApiProperty({required:false})
  drugs: Array<object>
}

File: prescription-service/src/main/java/com/albatha/nexthealth/prescription/graphql/model/PrescriptionDTO.java
package com.albatha.nexthealth.prescription.graphql.model;

import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class PrescriptionDTO {

	private String prescriptionNumber;
	private String recordNumber;
	private OffsetDateTime visitDate;
	private String diagnosis;
	private String pin;
	private PatientDTO patient;
	private DoctorDTO doctor;
	private FacilityDTO sendingFacility;
	private java.util.List<DrugDTO> drugs;
	private InsuranceDTO insurance;
	private OffsetDateTime createdDate;

}
